src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1299,1344 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                thisPLHP.fuelType = DataGlobalConstants::ResourceType::Natural_Gas;
+                errorsFound = true;
+                FuelTypeError = false;
+            }
+
+            // A7 End use category
+            thisPLHP.endUseSubcat = UtilityRoutines::MakeUPPERCase(fields.at("end_use_subcategory").get<std::string>());
+            if (thisPLHP.endUseSubcat == "") {
+                thisPLHP.endUseSubcat = "Heat Pump Fuel Fired"; // or "General"?
+            }
+
+            // N1 Nominal heating capacity
+            auto tmpRefCapacity = fields.at(refCapFieldTag);
+
+            if (tmpRefCapacity == "Autosize") {
+                thisPLHP.referenceCapacity = DataSizing::AutoSize;
+                thisPLHP.referenceCapacityWasAutoSized = true;
+            } else {
+                thisPLHP.referenceCapacity = tmpRefCapacity.get<Real64>();
+            }
+
+            // N2 Nominal heating capacity
+            thisPLHP.referenceCOP = fields.at("nominal_cop").get<Real64>();
+            if (thisPLHP.referenceCOP <= 0.0) thisPLHP.referenceCOP = 1.0;
+
+            // N3 Design flow rate
+            auto tmpFlowRate = fields.at("design_flow_rate");
+            if (tmpFlowRate == "Autosize") {
+                thisPLHP.loadSideDesignVolFlowRate = DataSizing::AutoSize;
+                thisPLHP.loadSideDesignVolFlowRateWasAutoSized = true;
+            } else {
+                thisPLHP.loadSideDesignVolFlowRate = tmpFlowRate.get<Real64>();
+            }
+
+            // GAHP: Add a default source side flow rate, not from input
+            Real64 defDummyASDesVolFlowRate = 1.0;
+            thisPLHP.sourceSideDesignVolFlowRate = defDummyASDesVolFlowRate;
+
+            // auto tmpSourceFlowRate = fields.at("source_side_reference_flow_rate");
+            // if (tmpSourceFlowRate == "Autosize") {
+            //    thisPLHP.sourceSideDesignVolFlowRate = DataSizing::AutoSize;
+            //    thisPLHP.sourceSideDesignVolFlowRateWasAutoSized = true;
+            //} else {
+            //    thisPLHP.sourceSideDesignVolFlowRate = tmpSourceFlowRate.get<Real64>();
+            //}
+
+            // N4 Design supply temperature
+            auto tmpDesSupTemp = fields.at("design_supply_temperature");
+            if (tmpDesSupTemp == "Autosize") {
+                // sizing
+            } else {
+                thisPLHP.desSupplyTemp = tmpDesSupTemp.get<Real64>();
+            }
+
+            // N5 Design temperature lift
+            auto tmpDesTempLift = fields.at("design_temperature_lift");
+            if (tmpDesTempLift == "Autosize") {
+                // sizing
+            } else {
+                thisPLHP.desTempLift = tmpDesTempLift.get<Real64>();
+            }
+
+            // N6 Sizing factor
+            if (fields.find("sizing_factor") != fields.end()) {
+                thisPLHP.sizingFactor = fields.at("sizing_factor").get<Real64>();
```

Comment:
Here's an `at` following a `find` again.

Reply:
Addressed in earlier changes.